src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs

```diff
@@ -39,25 +47,40 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
                 var connectionKey = $"{_connectionString}_{_queueName}";
                 if (!_queueClientConnections.TryGetValue(connectionKey, out var queueClient))
                 {
-                    queueClient = new QueueClient(_connectionString, _queueName,ReceiveMode.PeekLock, RetryPolicy.NoRetry);
+                    queueClient = new QueueClient(_connectionString, _queueName, ReceiveMode.PeekLock, RetryPolicy.NoRetry);
 
                     if (!_queueClientConnections.TryAdd(connectionKey, queueClient))
                     {
                         return new HealthCheckResult(context.Registration.FailureStatus, description: "New QueueClient connection can't be added into dictionary.");
                     }
                 }
 
+                // If the queue is not yet available, GetQueueAsync will throw and _requiresSession will not have any value
+                if (!_requiresSession.HasValue)
+                {
+                    var managementClient = new ManagementClient(_connectionString);
+                    var queueDescription = await managementClient.GetQueueAsync(_queueName);
+                    _requiresSession = queueDescription.RequiresSession;

```

Comment:
Well, the problem here is you use management to get information about if the queue require session, we prefer add new flag into AddXXX because query management is overhead  in this process

Reply:
My thought was this will be done only the first time, and hence the overhead should be fine - once the requiresSession is set, ManagementClient will not be used. The advantage is that the user does not have to explicitly pass a requiresSession, and can be figured out from the queue. However if you prefer passing it explicitly, I can either create an optional session argument defaulted to false in ctor, or a new AddSession() method.

Reply:
Sounds good option session defaulted to false, or some like Options to configure this like this

Reply:
Please tagme when push new commits on this to review and merge

Reply:
@unaizorrilla Can you please review? Thanks!